integration-tests/features/steps/user_tag.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""Tests For crowdsourcing API end-points."""
+import requests
+
+from behave import given, then, when
+from urllib.parse import urljoin
+
+from src.parsing import *
+from src.utils import *
+from src.authorization_tokens import *
+
+
+@given('Master tag list api is running')
+def running_master_tag_list_api(context):
+    """Wait for the master tag list REST API to be available."""
+    if not context.is_master_tag_list_service_running(context):
+        context.wait_for_master_tag_list_service(context, 60)
+
+
+def master_tag_list_url(context, ecosystem):
+    """Construct URL for master tag list REST API call."""
+    return urljoin(context.coreapi_url,
+                   'api/v1/master-tags/{e}'.format(e=ecosystem))
+
+
+def get_master_tag_list(context, ecosystem, use_token):
+    """Call API endpoint master tag list."""
+    if use_token:
+        context.response = requests.get(master_tag_list_url(context, ecosystem),
+                                        headers=authorization(context))
+    else:
+        context.response = requests.get(master_tag_list_url(context, ecosystem))
+
+
+@when("I access master tag list API point for ecosystem {ecosystem} without authorization token")
+def fetch_master_tag_list_without_token(context, ecosystem):
+    """Fetch master tag list for given ecosystem REST API call."""
+    get_master_tag_list(context, ecosystem, False)
+
+
+@when("I access master tag list API point for ecosystem {ecosystem} with authorization token")
+def fetch_master_tag_list_with_token(context, ecosystem):
+    """Fetch master tag list for given ecosystem REST API call."""
+    get_master_tag_list(context, ecosystem, True)
+
+
+@then('I should get 401 status code as response for master tag list')
+def check_status_code_for_master_tag_list(context):
+    """Check master tag list require authorization tokens."""
+    assert context.response.status_code == 401
+
+
+@then('I should get json object contains tag_list which is a array of strings')
+def check_master_tag_list_response_json(context):
+    """Check that json response contains master tags."""
+    json_data = context.response.json()
+    assert 'ecosystem' in json_data
+    assert 'tag_list' is not any(json_data.get('tag_list', []))
```

Comment:
I'm not sure what's checked here. Is it just a test for non empty list?

Reply:
typo: an array

Reply:
yes, It's a test for a non-empty tag list.